### ISSUE TITLE
Is Pi Node broken?

### ISSUE THREAD

**Issue by purupokharel** [pi-node/instructions#231]
Running Pi Node on a Mac mini (Late 2014) 2.8GHz Dual-Core Intel Core i5 with 8GB Memory and 1.5GB VRAM. 

Ducker version is 4.4.2 (73305) and Pi Network App version is 0.4.5(0.4.5) as of Jan 20, 2022. While NODE is launched, it passes through Ducker and Port testing, allows me to continue and stays forever in the White Screen with the message, and the message never goes away: 

**"Please wait... Loading may take up to one minute."** 

Does anyone have an idea why Pi Node is not operating? Whether anyone out there is having the same issue? I did not find it anywhere brought forward as an issue other than Reddit, where no reliable solution found. 

I verified that it is not a ducker issue since docker verifies all Pi open ports. There is no issue with ports being closed. In such a case, what might trigger the Pi Node to keep trying to establish its server resources, and it is not able to pull anything at all? The following is the screenshot where Pi Node is stuck with a spinning line with the aforementioned message in the quotes.

[Screen Shot 2022-01-20 at 6 30 37 PM](https://user-images.githubusercontent.com/98131008/150438627-0479376a-51b9-45c0-9f65-03512b8841dc.png)

**Comment by michelferia** [comment]
I have the same issue, seems that the test net is not operational, at least for new nodes...

**Comment by parknochang** [comment]
Infinite loading when installing a new node. The cause is an error creating the user-preferences.json file. It must be resolved by PCT.

**Comment by michelferia** [comment]
Hi @parknochang, do you know how to solve it? maybe editing the json file inside docker? thanks!

![node_issue](https://user-images.githubusercontent.com/11069310/150677628-912afb0a-32a4-4383-8a8b-42b75b22f3e5.png)

everything looks fine, open ports, docker running but Node App is stuck...
any idea how to fix it?
thanks again!

**Comment by MartinRe63** [comment]
I've the same issue here. My system can write the user-preferences.json file. I've removed it - afterwards the client created it again.  So access right's for the PI node are in place. My first mining session is also done - nothing changed.

**Comment by MartinRe63** [comment]
Maybe it's only possible if a pioneer has done it's first mining session? Is there any condition to get the node downloaded? Anyhow ... according to the info the testnet has 10.000 nodes + 100.000 nodes in wait mode ... nodes for the mainnet will be hosted by the kernel team only until the closed network phase has reached it's end. It's no drama for the PI blockchain, if a couple of nodes can't join. There seems to be a sufficient amount of nodes available for test purposes ...

**Comment by MartinRe63** [comment]
I've analysed the log file. It's starting with these errors and I would assume they also lead to the situation, that my node is not starting: 
[2022-01-24 13:43:53.797] [info] [PERSISTENT_STORE] INIT:  C:\Users\Martin\AppData\Roaming\Pi Network\user-preferences.json
[2022-01-24 13:43:53.813] [debug] parseDataFile: filePath C:\Users\Martin\AppData\Roaming\Pi Network\user-preferences.json  defaults { windowWidth: 800, windowHeight: 600 }
[2022-01-24 13:43:53.813] [debug] dockerContainerInitConfig: start
[2022-01-24 13:43:53.813] [debug] [PERSISTENT_STORE] get: dockerContainerVolume
[2022-01-24 13:43:53.813] [debug] defaultDirectory: start
[2022-01-24 13:43:53.813] [debug] defaultDirectory: dir: C:\Users\Martin\AppData\Roaming\Pi Network\docker_volumes
[2022-01-24 13:43:53.813] [warn] dockerContainerInitConfig: fall back to user default directory:  C:\Users\Martin\AppData\Roaming\Pi Network\docker_volumes
[2022-01-24 13:43:53.813] [debug] detectOS: start
[2022-01-24 13:43:53.829] [info] MAIN: starting:  Proton
[2022-01-24 13:43:53.829] [info] MAIN: env production
[2022-01-24 13:43:53.875] [debug] [PERSISTENT_STORE] get: uuid
[2022-01-24 13:43:53.891] [debug] [PERSISTENT_STORE] get: uuid
[2022-01-24 13:43:53.891] [debug] [PERSISTENT_STORE] get: uuid
[2022-01-24 13:43:53.891] [debug] [PERSISTENT_STORE] get: switchStatus
[2022-01-24 13:43:53.891] [debug] [PERSISTENT_STORE] get: switchStatus
[2022-01-24 13:43:54.006] [info] MAIN: createWindow
[2022-01-24 13:43:54.007] [debug] initMainWindow: start
[2022-01-24 13:43:54.050] [debug] initMainWindow: open React app at  C:\Users\Martin\AppData\Local\Programs\pi-network-desktop\resources\app.asar\app\app.html
[2022-01-24 13:43:54.065] [debug] [PERSISTENT_STORE] get: switchStatus
[2022-01-24 13:43:54.143] [info] Checking for update
[2022-01-24 13:43:54.143] [info] initOsStats: start
[2022-01-24 13:43:54.281] [info] initHeartbeat: start
[2022-01-24 13:43:54.291] [info] initStellarConfigurationUpdate: start
[2022-01-24 13:43:54.291] [info] initStellarConfigurationUpdate: start
**[2022-01-24 13:43:54.373] [error] run: stats: drive.info(): err: Error: disk name invalid and / not found
    at C:\Users\Martin\AppData\Local\Programs\pi-network-desktop\resources\app.asar\node_modules\node-os-utils\lib\drive.js:62:17
[2022-01-24 13:43:55.179] [error] run: stellar-config: request error:  {**
  message: 'certificate has expired',
  name: 'Error',
  stack: 'Error: certificate has expired\n' +
    '    at TLSSocket.onConnectSecure (_tls_wrap.js:1321:34)\n' +
    '    at TLSSocket.emit (events.js:223:5)\n' +
    '    at TLSSocket.EventEmitter.emit (domain.js:475:20)\n' +
    '    at TLSSocket._finishInit (_tls_wrap.js:794:8)\n' +
    '    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:608:12)',
  config: {
    url: 'https://sync.minepi.com/stellar-configuration',
    method: 'get',

**Comment by michelferia** [comment]
Maybe the problem is coming from the certificate? The real issue here is the non-reply from PCT or other devs....

**Comment by parknochang** [comment]
The problem is with the pi network node server. The new node has missing information in the user-preferences.json file. This will be fixed in the PCT. Wait.

**Comment by parknochang** [comment]
I know how, but there can be a penalty for editing the file. Be careful.

**Comment by michelferia** [comment]
Thanks for your info, I will wait so, good day!!

**Comment by MartinRe63** [comment]
Thx parknochang. What is the meaning of PCT and how can I check if this is fixed? Is there any official release list of the node or something where the community can see about the progress on this? Or can you see the progress and will close the ticket here, once the issue is solved? Thx.

**Comment by steeveee100** [comment]
PCT means Pi Core Team! That's it.

**Comment by purupokharel** [comment]
Today, my issue got resolved by device specific port-forwarding. I reserved the Device IP, and forwarded the range of ports (31400-31409) to that specific computer running the Pi Node. This resolved the issue immediately. If you are running multiple Pi Nodes, this solution might not work for you. If it was a coincident, then it might be that the PCT got the attention, and resolved on their end by the time I was trying to isolate and resolve it today. 

If you are still having the issue, I would suggest to shut down your computer, wait for a while and turn back on to see if it gets resolved. If it was the PCT that paid attention to this issue and resolved it, many thanks to the devs. Thank you all!
